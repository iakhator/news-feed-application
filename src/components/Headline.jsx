import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import SortBy from './SortBy';

/**
 * Renders the articles from the news api
 * @param {Array} props: data passed from Newsheadline
 * @returns {void}
 */
function Headline(props) {
  const { newsArticle, sourceId, sortBy } = props;
  return (
    <div>
      <div className="sort-header sort-fixed affix">
        <SortBy sourceId={sourceId} />
        <h2 className="article-title">
          {sortBy.toUpperCase()} {sourceId.replace(/-/g, ' ').toUpperCase()}{' '}
          NEWS
        </h2>
      </div>
      {newsArticle.map(article => (
        <div className="articles" key={article.title}>
          <div className="row single">
            <div className="col-md-4">
              <img className="img-responsive" src={article.urlToImage} alt="" />
            </div>
            <div className="col-md-8">
              <h3 className="page-header">
                <Link to={article.url} target="_blank">
                  {article.title}
                </Link>
              </h3>
              <div>
                <p className="description page-header">
                  <i className="fa fa-newspaper-o" aria-hidden="true" />{' '}
                  {article.description}...
                </p>
                <p className="pull-left publish-time">
                  <i className="fa fa-clock-o" aria-hidden="true" />
                  {article.publishedAt}
                </p>
                <a
                  className="btn btn-sm see-more pull-right"
                  href={article.url}
                  target="_blank"
                >
                  See more yes
                </a>
              </div>
            </div>
          </div>
        </div>
      ))}
    </div>
  );
}

Headline.propTypes = {
  newsArticle: PropTypes.array.isRequired,
  sourceId: PropTypes.string.isRequired,
  sortBy: PropTypes.string.isRequired
};

export default Headline;
