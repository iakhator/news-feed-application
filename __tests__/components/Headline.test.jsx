import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Headline from '../../src/components/Headline';

describe('NewsHeadline', () => {
  const props = {
    newsArticle: [],
    sourceId: 'cnn'.toUpperCase(),
    sortBy: 'top'.toUpperCase()
  };

  const container = shallow(<Headline {...props} />);

  it('should render correctly', () => {
    const tree = renderer.create(
      <Headline {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have sort-header and article-title class', () => {
    const wrapper = mount(<Headline {...props} />);
    expect(wrapper.find('.sort-header').length).toBe(1);
    expect(wrapper.find('.article-title').length).toBe(1);
  });

  it('Should render <div /> tags', () => {
    expect(container.find('div').toExist);
  });
});
