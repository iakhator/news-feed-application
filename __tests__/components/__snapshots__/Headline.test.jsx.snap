// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsHeadline should render correctly 1`] = `
<div>
  <div
    className="sort-header sort-fixed affix"
  >
    <div>
      <ul
        className="sort pull-left"
      />
    </div>
    <h2
      className="article-title"
    >
      TOP
       
      CNN
       
      NEWS
    </h2>
  </div>
</div>
`;
